src/platforms/native/meson.build

```diff
@@ -39,6 +39,14 @@ probe_native_args = [
 	'-DBLACKMAGIC',
 ]
 
+trace_protocol = get_option('trace_protocol')
+if trace_protocol == '1'
```

Comment:
Due to the default being '3' and this intending to be "whatever the platform's default would be", this check needs to change slightly to `trace_protocol in ['1', '3']` to accommodate that and result in a net-zero change on the OOBE (Out Of Box Experience).

We suggest indenting the line below and changing the f-string to having a constant `=1` as a result.

Reply:
I agree the OOB experience is paramount!

Isn't the default trace protocol OOB better set in the crossfile?

Reply:
No, as the cross-file is only a suggestion - there's absolutely nothing stopping a user from doing `meson setup build --cross-file=cross-file/arm-none-eabi.ini -Dprobe=native` and that needs to work, so all defaults need encoding directly in the build system. This is related to the same problem as a user reconfiguring an existing build directory to a different probe (eg, `meson setup build --reconfigure -Dprobe=stlink` having been configured for native) - this is a perfectly reasonable thing to want to do and needs to work as well but never re-reads any cross files to change configuration options.